projection/ibc_channel/view/ibc_channels.go

```diff
@@ -564,6 +565,167 @@ func (ibcChannelsView *IBCChannels) List(
 	return channels, paginationResult, nil
 }
 
+func (ibcChannelsView *IBCChannels) ListChannelsGroupByChainId(
+	order IBCChannelsListOrder,
+	filter IBCChannelsListFilter,
+	pagination *pagination.Pagination,
+) (
+	[]ChainChannels,
+	*pagination.PaginationResult,
+	error,
+) {
+
+	stmtBuilder := ibcChannelsView.rdb.StmtBuilder.Select(
+		"channel_id",
+		"port_id",
+		"connection_id",
+		"counterparty_channel_id",
+		"counterparty_port_id",
+		"counterparty_chain_id",
+		"status",
+		"packet_ordering",
+		"last_in_packet_sequence",
+		"last_out_packet_sequence",
+		"total_transfer_in_count",
+		"total_transfer_out_count",
+		"total_transfer_out_success_count",
+		"total_transfer_out_success_rate",
+		"created_at_block_time",
+		"created_at_block_height",
+		"verified",
+		"description",
+		"last_activity_block_time",
+		"last_activity_block_height",
+		"bonded_tokens",
+	).From(
+		"view_ibc_channels",
+	)
+
+	if filter.MaybeStatus != nil {
+		if *filter.MaybeStatus {
+			// Filtered channels in `opened` status
```

Comment:
Just a suggestion, again this comment can be delivered with code. Not pushing for a change, but just open this for discussion for future. For example
```
stmtBuilder = addOpenedStatusFilterConditionTo(stmtBuilder)
```

Reply:
Thanks for the advice! Yes, this will be easier to read! Let me update this part.